updater/lib/dependabot/updater/group_update_creation.rb

```diff
@@ -0,0 +1,239 @@
+# frozen_string_literal: true
+
+require "dependabot/dependency_change_builder"
+
+# This module contains the methods required to build a DependencyChange for
+# a single DependencyGroup.
+#
+# When included in an Operation it expects the following to be available:
+# - job: the current Dependabot::Job object
+# - dependency_snapshot: the Dependabot::DependencySnapshot of the current
+#   repo state
+# - error_handler: a Dependabot::UpdaterErrorHandler to report
+#
+module Dependabot
+  class Updater
+    module GroupUpdateCreation
```

Comment:
I considered two alternative approaches:
- Let's make a GroupUpdate base class!
- Let's create a new GroupUpdaterBuilder object we instantiate and call!

I ended up 👎🏻 on both of those as I'd rather not add inherence into the Updater given the problems we've had with branching code and I wasn't totally sold on the value of a `GroupUpdaterBuilder` object as a separate component we might want to add tests on specifically.

I ended up defaulting to a coarse composition approach with a view that we can revisit this later as there's still some DRYing out and test improvement to be done on the other Operations and the right pattern is likely to become clearer with more examples.